R/as_polars.R

```diff
@@ -0,0 +1,156 @@
+#' To polars DataFrame
+#'
+#' [as_polars_df()] is a generic function that converts an R object to a
+#' polars DataFrame. It is basically a wrapper for [pl$DataFrame()][pl_DataFrame],
+#' but has special implementations for Apache Arrow-based objects such as
+#' polars [LazyFrame][LazyFrame_class] and [arrow::Table].
+#'
+#' For [LazyFrame][LazyFrame_class] objects, this function is a shortcut for
+#' [$collect()][LazyFrame_collect] or [$fetch()][LazyFrame_fetch], depending on
+#' whether the number of rows to fetch is infinite or not.
+#' @rdname as_polars_df
+#' @param x Object to convert to a polars DataFrame.
+#' @param ... Additional arguments passed to methods.
+#' @examplesIf requireNamespace("arrow", quietly = TRUE)
+#' at = arrow::as_arrow_table(mtcars)
+#'
+#' # Convert an arrow Table to a polars LazyFrame
+#' lf = as_polars_df(at)$lazy()
+#'
+#' # Collect all rows
+#' as_polars_df(lf)
+#'
+#' # Fetch 5 rows
+#' as_polars_df(lf, 5)
+#' @export
+as_polars_df = function(x, ...) {
+  UseMethod("as_polars_df")
+}
+
+
+#' @rdname as_polars_df
+#' @export
+as_polars_df.default = function(x, ...) {
+  as_polars_df(as.data.frame(x, stringsAsFactors = FALSE), ...)
+}
+
+
+#' @rdname as_polars_df
+#' @export
+as_polars_df.data.frame = function(x, ...) {
+  pl$DataFrame(x)
+}
+
+
+#' @rdname as_polars_df
+#' @export
+as_polars_df.DataFrame = function(x, ...) {
+  x
+}
+
+
+#' @rdname as_polars_df
+#' @export
+as_polars_df.GroupBy = function(x, ...) {
+  x$to_data_frame()
+}
```

Comment:
Sorry, this was incorrect.......
Not polars DataFrame, but R data.frame.
I will open a follow-up PR.

Reply:
#522